pilot/cmd/pilot-agent/status/server.go

```diff
@@ -57,9 +57,9 @@ var (
 
 // KubeAppProbers holds the information about a Kubernetes pod prober.
 // It's a map from the prober URL path to the Kubernetes Prober config.
-// For example, "/app-health/hello-world/livez" entry contains livenss prober config for
+// For example, "/app-health/hello-world/livez" entry contains liveness prober config for
 // container "hello-world".
-type KubeAppProbers map[string]*corev1.HTTPGetAction
+type KubeAppProbers map[string]*corev1.Probe
```

Comment:
Prefer declaring istio specific struct, actually k8s Probe contains too many useless fields

Reply:
The only useless fields that are in this struct are the `ExecAction` and `TCPSocketAction`, other than that I'm fairly confident that staying as close to user's intentions are the way to go. 

We could create a new struct for this, but that'd basically be `corev1.Probe` with the `Handler` hardcoded to `HTTPGetAction`.

Reply:
```
type Probe struct {
	// The action taken to determine the health of a container
	Handler `json:",inline" protobuf:"bytes,1,opt,name=handler"`
	// Number of seconds after the container has started before liveness probes are initiated.
	// More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes
	// +optional
	InitialDelaySeconds int32 `json:"initialDelaySeconds,omitempty" protobuf:"varint,2,opt,name=initialDelaySeconds"`
	// Number of seconds after which the probe times out.
	// Defaults to 1 second. Minimum value is 1.
	// More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes
	// +optional
	TimeoutSeconds int32 `json:"timeoutSeconds,omitempty" protobuf:"varint,3,opt,name=timeoutSeconds"`
	// How often (in seconds) to perform the probe.
	// Default to 10 seconds. Minimum value is 1.
	// +optional
	PeriodSeconds int32 `json:"periodSeconds,omitempty" protobuf:"varint,4,opt,name=periodSeconds"`
	// Minimum consecutive successes for the probe to be considered successful after having failed.
	// Defaults to 1. Must be 1 for liveness and startup. Minimum value is 1.
	// +optional
	SuccessThreshold int32 `json:"successThreshold,omitempty" protobuf:"varint,5,opt,name=successThreshold"`
	// Minimum consecutive failures for the probe to be considered failed after having succeeded.
	// Defaults to 3. Minimum value is 1.
	// +optional
	FailureThreshold int32 `json:"failureThreshold,omitempty" protobuf:"varint,6,opt,name=failureThreshold"`
}

```

We only need timeout and the HTTPGetAction?

Reply:
Wouldn't you want the rest of the parameters as well? After all the probes are executed by the Kubelet and all other parameters would have to be moved to the pilot-agent as well to keep the user's intentions.

Reply:
nope, pilot-agent will never need `InitialDelaySeconds ` `FailureThreshold ` `PeriodSeconds `, it is transparent

Reply:
I was under the assumption that the entire `httpGet` block was moved away from the original container, and added to the pilot-agent, but figured out I was wrong now. If I understand correctly, the readiness/liveness check is kept on the original, and what happens here is that the original path and port is set to an environment variable for the pilot-agent to pick up, and the original readiness/liveness check is rewritten to have the pilot-agent "proxy" path that exposes the endpoint in plaintext. Correct me if I'm wrong, please.

I'll create a patch this week to fix this with an Istio specific struct to pass to the pilot-agent.

Reply:
yeah, you don't need to pass these information other than timeout/path/port to pilot agent, but letting kubelet retain the original timeout and retry, initial delay on the podspec is helpful as you mentioned.